source/common/upstream/load_balancer_impl.cc

```diff
@@ -27,13 +27,19 @@ LoadBalancerBase::LoadBalancerBase(const HostSet& host_set, const HostSet* local
         [this](const std::vector<HostSharedPtr>&, const std::vector<HostSharedPtr>&) -> void {
           regenerateZoneRoutingStructures();
         });
-    local_host_set_->addMemberUpdateCb(
+    local_host_set_member_update_cb_handle_ = local_host_set_->addMemberUpdateCb(
         [this](const std::vector<HostSharedPtr>&, const std::vector<HostSharedPtr>&) -> void {
           regenerateZoneRoutingStructures();
         });
   }
 }
 
+LoadBalancerBase::~LoadBalancerBase() {
+  if (local_host_set_member_update_cb_handle_ != nullptr) {
+    local_host_set_->removeMemberUpdateCb(local_host_set_member_update_cb_handle_);
```

Comment:
Wondering if we could use RAII on the handles to have them clean themselves up on object destruction..